collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/EntityFormTest.kt

```diff
@@ -5,27 +5,162 @@ import org.junit.Rule
+
+            .startBlankForm("One Question Entity Update")
+            .assertQuestion("Select person")
+            .assertText("Roman Roy")
+            .assertText("Logan Roy")
+    }
+
+    @Test
+    fun fillingEntityUpdateForm_updatesEntityForFollowUpForms() {
+        testDependencies.server.addForm("one-question-entity-update.xml", listOf("people.csv"))
+
+        rule.withMatchExactlyProject(testDependencies.server.url)
+            .enableLocalEntitiesInForms()
+
+            .startBlankForm("One Question Entity Update")
+            .assertQuestion("Select person")
+            .clickOnText("Roman Roy")
+            .swipeToNextQuestion("Name")
+            .answerQuestion("Name", "Romulus Roy")
+            .swipeToEndScreen()
+            .clickFinalize()
+
+            .startBlankForm("One Question Entity Update")
+            .assertText("Romulus Roy")
+            .assertTextDoesNotExist("Roman Roy")
+    }
+
+    @Test
+    fun fillingEntityFollowUpForm_whenOnlineDuplicateHasHigherVersion_showsOnlineVersion() {
+        testDependencies.server.addForm("one-question-entity-update.xml", listOf("people.csv"))
+
+        val mainMenuPage = rule.withMatchExactlyProject(testDependencies.server.url)
+            .enableLocalEntitiesInForms()
+
+            .startBlankForm("One Question Entity Update")
+            .assertQuestion("Select person")
+            .clickOnText("Roman Roy")
+            .swipeToNextQuestion("Name")
+            .answerQuestion("Name", "Romulus Roy")
+            .swipeToEndScreen()
+            .clickFinalize()
+
+        testDependencies.server.updateMediaFile(
+            "one-question-entity-update.xml",
+            "people.csv",
+            "updated-people.csv"
+        )
+
+        mainMenuPage.clickFillBlankForm()
+            .clickRefresh()
+            .clickOnForm("One Question Entity Update")
+            .assertText("Ro-Ro Roy")
+            .assertTextDoesNotExist("Romulus Roy")
+            .assertTextDoesNotExist("Roman Roy")
+    }
+
+    @Test
+    fun fillingEntityFollowUpForm_whenOfflineDuplicateHasHigherVersion_showsOfflineVersion() {
+        testDependencies.server.addForm(
+            "one-question-entity-update.xml",
+            mapOf("people.csv" to "updated-people.csv")
+        )
+
+        rule.withMatchExactlyProject(testDependencies.server.url)
+            .enableLocalEntitiesInForms()
+
+            .startBlankForm("One Question Entity Update")
+            .assertQuestion("Select person")
+            .clickOnText("Ro-Ro Roy")
+            .swipeToNextQuestion("Name")
+            .answerQuestion("Name", "Romulus Roy")
+            .swipeToEndScreen()
+            .clickFinalize()
+
+            .startBlankForm("One Question Entity Update")
+            .assertText("Romulus Roy")
+            .assertTextDoesNotExist("Ro-Ro Roy")
+    }
+
+    @Test
+    fun fillingEntityUpdateForm_whenFormDoesNotUpdateLabel_updatesEntityForFollowUpForms() {
```

Comment:
Is this title correct? As I can see there is no update the name was `Roman Roy` and it is still `Roman Roy`.